### DISCUSSION TITLE
Jetstream getting error after changing consumer to durable

### DISCUSSION THREAD

**Discussion by alberk8** [nats-io/nats.net#564] [Q&A]
I am get this error when publishing to a existing stream where the consumer is changed to durable.  I end up deleting the Stream to get it working again. 

Server: 2.10.17
Client: 2.3.2

`[07:28:21.519  DBG][] NATS.Client.JetStream.NatsJSContext: No response received, retrying 1/2`

**Comment by mtmk** [comment]
hi @alberk8, you should not change consumers durability instead create a new one. could you share some code to explain the use case a bit more please?

**Reply by alberk8** [reply]
I have a Consumer that was ephemeral initially then I change the same Consumer to Durable.  There was no error or warning. So I assume it was good.  The problem only crop up when another program begins to publish to the same Stream is hitting No Responder error.   

I would be a great help if there is error message when there is a change of Consumer from ephemeral to durable and vice versa.

**Reply by mtmk** [reply]
those problems seems unrelated at first sight i.e. consumer update should not affect the stream publishing.

**Reply by alberk8** [reply]
I will do more testing and see what can cause the issue.  In the meantime, I find that the `Name `and `DurableName` should be set should be private set as it would cause runtime error if those mismatch.

```
 var consumerConfig = new ConsumerConfig("TestConsumer2")
 {
     //Name = "TestConsumer2",
     //DurableName = "TestConsumer3",
     FilterSubject = "TestSubject",
     DeliverPolicy = ConsumerConfigDeliverPolicy.All,
 };
```

**Reply by mtmk** [reply]
yes this is a known issue please feel free to create issue/PR 💯 cc @scottf
